controller/internal/routes/identity_api_model.go

```diff
@@ -34,357 +36,246 @@ const (
-	Type           *string                `json:"type"`
-	IsAdmin        *bool                  `json:"isAdmin"`
-	RoleAttributes []string               `json:"roleAttributes"`
-}
+func (factory *IdentityLinkFactoryImpl) Links(entity models.Entity) rest_model.Links {
+	links := factory.BasicLinkFactory.Links(entity)
+	links[EntityNameEdgeRouterPolicy] = factory.NewNestedLink(entity, EntityNameEdgeRouter)
+	links[EntityNameEdgeRouterPolicy] = factory.NewNestedLink(entity, EntityNameEdgeRouter)
 
-func NewIdentityApiCreate() *IdentityApiCreate {
-	f := false
-	return &IdentityApiCreate{
-		IsAdmin: &f,
-	}
+	return links
 }
 
-func (i *IdentityApiCreate) ToModel() (*model.Identity, []*model.Enrollment) {
-	identity := &model.Identity{}
-	identity.Name = stringz.OrEmpty(i.Name)
-	identity.IdentityTypeId = stringz.OrEmpty(i.Type)
-	identity.IsDefaultAdmin = false
-	identity.IsAdmin = boolOrFalse(i.IsAdmin)
-	identity.RoleAttributes = i.RoleAttributes
-	identity.Tags = i.Tags
-
+func MapCreateIdentityToModel(identity *rest_model.IdentityCreate, identityTypeId string) (*model.Identity, []*model.Enrollment) {
 	var enrollments []*model.Enrollment
-	if caId, ok := i.Enrollment[persistence.MethodEnrollOttCa]; ok {
-		caId := caId.(string)
-		enrollment := &model.Enrollment{
-			Method: persistence.MethodEnrollOttCa,
-			CaId:   &caId,
-		}
-		enrollments = append(enrollments, enrollment)
-	}
 
-	if _, ok := i.Enrollment[persistence.MethodEnrollOtt]; ok {
-		enrollments = append(enrollments, &model.Enrollment{
-			Method: persistence.MethodEnrollOtt,
-		})
+	ret := &model.Identity{
+		BaseEntity: models.BaseEntity{
+			Tags: identity.Tags,
+		},
+		Name:           stringz.OrEmpty(identity.Name),
+		IdentityTypeId: identityTypeId,
+		IsDefaultAdmin: false,
+		IsAdmin:        *identity.IsAdmin,
+		RoleAttributes: identity.RoleAttributes,
 	}
 
-	if val, ok := i.Enrollment[persistence.MethodEnrollUpdb]; ok {
-		username := val.(string)
-		enrollments = append(enrollments, &model.Enrollment{
-			Method:   persistence.MethodEnrollUpdb,
-			Username: &username,
-		})
+	if identity.Enrollment != nil {
+		if identity.Enrollment.Ott {
+			enrollments = append(enrollments, &model.Enrollment{
+				BaseEntity: models.BaseEntity{},
+				Method:     persistence.MethodEnrollOtt,
+				Token:      uuid.New().String(),
+			})
+		} else if identity.Enrollment.Ottca != "" {
+			caId := identity.Enrollment.Ottca.String()
+			enrollments = append(enrollments, &model.Enrollment{
+				BaseEntity: models.BaseEntity{},
+				Method:     persistence.MethodEnrollOttCa,
+				Token:      uuid.New().String(),
+				CaId:       &caId,
+			})
+		} else if identity.Enrollment.Updb != "" {
+			username := identity.Enrollment.Updb
+			enrollments = append(enrollments, &model.Enrollment{
+				BaseEntity: models.BaseEntity{},
+				Method:     persistence.MethodEnrollUpdb,
+				Token:      uuid.New().String(),
+				Username:   &username,
+			})
+		}
```

Comment:
should there be a final else and log to catch "just in case" situations should one ever arise?

Reply:
No, the else is that the identity is being created w/o enrollments.